rust/ql/src/queries/unusedentities/UnreachableCode.ql

```diff
@@ -12,19 +12,29 @@ import rust
 import codeql.rust.controlflow.ControlFlowGraph
 import codeql.rust.controlflow.internal.ControlFlowGraphImpl as ControlFlowGraphImpl
 
+/**
+ * Holds if `n` is an AST node that's unreachable.
+ */
+private predicate unreachable(AstNode n) {
+  not n = any(CfgNode cfn).getAstNode()
```

Comment:
We also need to restrict this to nodes that can possibly be part of a CFG, for example `ParenExpr`s cannot. I think the best heuristic is simply `n instanceof ControlFlowTree`.

Reply:
`n instanceof ControlFlowGraphImpl::ControlFlowTree` doesn't work ... unless this fix requires more of your work-in-progress to apply?

Reply:
Right. Try with
```ql
  exists(ControlFlowGraphImpl::ControlFlowTree cft |
    cft.succ(n, _, _)
    or
    cft.succ(_, n, _)
  )
```
instead.